src/step.cpp

```diff
@@ -133,65 +136,119 @@ void fields_chunk::phase_material(int phasein_time) {
 void fields::step_boundaries(field_type ft) {
   connect_chunks(); // re-connect if !chunk_connections_valid
 
-  // Do the metals first!
-  for (int i = 0; i < num_chunks; i++)
-    if (chunks[i]->is_mine()) chunks[i]->zero_metal(ft);
+  {
+    // Initiate receive operations as early as possible.
+    std::unique_ptr<comms_manager> manager = create_comms_manager();
 
-  /* Note that the copying of data to/from buffers is order-sensitive,
-     and must be kept consistent with the code in boundaries.cpp.
-     In particular, we require that boundaries.cpp set up the connections
-     array so that all of the connections for process i come before all
-     of the connections for process i' for i < i'  */
+    const auto &sequence = comms_sequence_for_field[ft];
+    for (const comms_operation &op : sequence.receive_ops) {
+      if (chunks[op.other_chunk_idx]->is_mine()) {
+        continue;
+      }
+      manager->receive_real_async(comm_blocks[ft][op.pair_idx], static_cast<int>(op.transfer_size),
```

Comment:
Maybe the `receive_real_async ` function should do the `static_cast<int>`, i.e. pass a `size_t` here, so that this code doesn't have to know that MPI is restricted to `int` sizes?

Reply:
Updated.